projects/plugins/wpcomsh/custom-colors/colors.php

```diff
@@ -397,10 +397,17 @@ public static function register_scripts_and_styles() {
 	/**
 	 * Add a 'custom-colors' body class to blogs with Custom Colors active.
 	 *
-	 * @param array $classes the array of classes to add custom class to.
+	 * @param ?array $classes the array of classes to add custom class to.
 	 */
 	public static function body_class( $classes ) {
-		array_push( $classes, 'custom-colors' );
+
+		// Safeguard against values that are not null, but not array either.
```

Comment:
I think you're worrying a little too much here. The code in core looks like this:
```php
    $classes = array_map( 'esc_attr', $classes );

    /**
     * Filters the list of CSS body class names for the current post or page.
     *
     * @since 2.8.0
     *
     * @param string[] $classes   An array of body class names.
     * @param string[] $css_class An array of additional class names added to the body.
     */
    $classes = apply_filters( 'body_class', $classes, $css_class );

    return array_unique( $classes );
```
So I think we'd be safe enough with leaving line 400 as-is and just doing like this in the body, since Core itself isn't going to handle any non-array correctly either.
```php
// Work around other plugins being stupid.
if ( ! is_array( $classes ) ) {
    $classes = array();
}
```
OTOH, even that's more than will be needed to fix the actual error we're seeing. Just the one line changing the `array_push` to `$classes[]` is enough for that.

Reply:
Yes, I guess you're right, if Core doesn't handle non-arrays, we should be good. Thanks!